### DISCUSSION TITLE
Add header before executing any function in Flutter.

### DISCUSSION THREAD

**Discussion by sabikrahat** [pocketbase/pocketbase#2280] [Q&A]
I want to pass every request with this header 
```
headers: {
          "Accept": "application/json",
          "Access-Control_Allow_Origin": "*"
        }
```

Is it possible? If yes how can I do this?  @ganigeorgiev 

It's really important because, in the release version of Flutter web, it shows errors of CORS issues. :(

**Comment by ganigeorgiev** [comment]
There is no global hook to modify the request data that is sent with the request, but I don't think you need it.

`Access-Control-Allow-Origin` is a **response** header (_aka. the server instructs the browser to allow requests from the specified origin_) and it doesn't make much sense sending it with the request.

PocketBase by default allows all origins, unless you've explicitly started the executable with `--origins`, so you shouldn't encounter any CORS issues. If that's not the case, then:

- check if you have a browser extension that could interfere with the response headers

- if you have a reverse proxy (eg. nginx), check its configuration and whether it is not interfering with the proxied PocketBase responses (you can verify it with the browser's Inspect tool > Network > Headers tab > Response headers)

- ensure that the PocketBase server is accessible from the Flutter app (network errors could result in CORS issues since the preflight/`OPTIONS` request will fail)


If none of the above helps pinpointing the issue, please try to create a minimal reproducible repo and I'll investigate it (_for me it works fine testing it locally with the default app generated with `flutter create myapp`_).
